test/pprof.cc

```diff
@@ -0,0 +1,93 @@
+#include <catch2/catch.hpp>
+
+extern "C" {
+#include "native/pprof.h"
+}
+
+const char *empty_string = "";
+
+// cannot use null; use empty_string instead
```

Comment:
Right, this is kind of annoying.  I think the correct semantics for the string table are to intern NULL as the empty string.  I'll add it to my to-do (I went back and forth on this issue, so it's possible there's even a commented-out line implementing it).

Reply:
Yeah, avoiding it at the lower level forces all callers to deal with it, and if they _don't_ then there is now a failure condition that has to be resolved. If the lower level just handles it then there isn't an error condition anymore, and makes it more convenient for the caller. I think it's the right move, anyway.

Reply:
One potential issue is if the user sends NULL, but the size is nonzero.  There are two obvious answers.
1.  It's an error.  Return.
2.  It's harmless.  Internally set the string to `""` and the size to 0, no matter the presented size.

It is slightly annoying for a library to be overly forgiving, which tends to delocalize (in code) the presentation of an error from its origin, but in the average case I expect a NULL string to be semantically equivalent to the empty string for the purpose of this library consumers.  I'll go with 2.

Reply:
If the pointer is null and the size is non-zero, then there is definitely a bug in the calling code. To some degree, garbage in, garbage out is reasonable. In certain C libraries I'd advocate for an `assert()` only. The size is authoritative, and if they give us a nullptr with non-zero size then that's a bug in the caller; the assert is just there to help them find their own bugs.

What do you think about having an `assert` for debug builds which would catch the error, but then for production builds it coerces to the empty string in such cases? Trying to get the best of both worlds.

Reply:
That's a great idea.  It's a zero cost safety net and will probably help us catch our own bugs.